generators/server/templates/src/test/java/package/web/rest/KafkaResourceIT.java.ejs

```diff
@@ -18,115 +18,146 @@
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-<%_ if (cacheProvider === 'redis') { _%>
-import org.junit.jupiter.api.extension.ExtendWith;
-<%_ } _%>
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.test.annotation.DirtiesContext;
 import org.springframework.test.web.servlet.MockMvc;
 import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.util.ReflectionUtils;
 import org.testcontainers.containers.KafkaContainer;
 
+import java.lang.reflect.Field;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.HashMap;
 import java.util.Map;
+import java.util.Set;
 
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
-<%_ if (authenticationType === 'oauth2') { _%>
-@SpringBootTest(classes = {<%= mainClass %>.class, TestSecurityConfiguration.class})
-<%_ } else if (authenticationType === 'uaa' && applicationType !== 'uaa') { _%>
-@SpringBootTest(classes = {SecurityBeanOverrideConfiguration.class, <%= mainClass %>.class})
-<%_ } else { _%>
-@SpringBootTest(classes = <%= mainClass %>.class)
-<%_ } _%>
-@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_CLASS)
-<%_ if (cacheProvider === 'redis') { _%>
-@ExtendWith(RedisTestContainerExtension.class)
-<%_ } _%>
-public class <%= upperFirstCamelCase(baseName) %>KafkaResourceIT {
-
-    private MockMvc restMockMvc;
+class <%= upperFirstCamelCase(baseName) %>KafkaResourceIT {
 
     private static boolean started = false;
-
     private static KafkaContainer kafkaContainer;
 
-    @Autowired
-    private <%= upperFirstCamelCase(baseName) %>KafkaProducer producer;
-
-    @Autowired
-    private <%= upperFirstCamelCase(baseName) %>KafkaConsumer consumer;
-
-    private static final int MAX_ATTEMPT = 5;
+    private <%= upperFirstCamelCase(baseName) %>KafkaResource kafkaResource;
+    private MockMvc restMockMvc;
 
     @BeforeAll
-    public static void startServer() {
+    static void startServer() {
         if (!started) {
             startTestcontainer();
             started = true;
         }
     }
 
     private static void startTestcontainer() {
-        kafkaContainer = new KafkaContainer("<%= KAFKA_VERSION %>");
+        kafkaContainer = new KafkaContainer("5.3.1");
```

Comment:
we can put <%= KAFKA_VERSION %> instead of 5.3.1 ?

Reply:
:+1:

Reply:
done